### PR TITLE
drawio: build from source

### PR DESCRIPTION
**Pull request by alyssais**
###### Description of changes

This allows using drawio on other architectures, and on Darwin. The implementation is adapted from the podman-desktop package.

I only use this package through the CLI, so it would be really useful if somebody could check that all the graphical stuff is working as expected.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/drawio/default.nix
Comment (DarkOnion0) [inline pkgs/applications/graphics/drawio/default.nix:4]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, fetchYarnDeps
, makeDesktopItem
, copyDesktopItems
, desktopToDarwinBundle
, fixup_yarn_lock
, makeWrapper
, nodejs
, yarn
, electron
```

I suggest put each attribute in a single line here. It helps in future diffs.

File: pkgs/applications/graphics/drawio/default.nix
Comment (DarkOnion0) [inline pkgs/applications/graphics/drawio/default.nix:25]:
```suggestion
    copyDesktopItems
    fixup_yarn_lock
    makeWrapper
    nodejs
    yarn
```

Same as above, it should be better for future diffs and readability

File: pkgs/applications/graphics/drawio/default.nix
Comment (DarkOnion0) [inline pkgs/applications/graphics/drawio/default.nix:1]: Thanks a lot for the PR, I was planning to port it during the summer, but your help is welcome :hugs: 

Apps run and works as expected. The GUI is working fine too, even though I get weird error message that I don't have with the current package
```bash
did not find extension DRI_Mesa version 1
failed to bind extensions
MESA-LOADER: failed to open zink: /run/opengl-driver/lib/dri/zink_dri.so: cannot open shared object file: No such file or directory (search paths /run/opengl-driver/lib/dri, suffix _dri)
```

Result of `nixpkgs-review pr 228907` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>drawio</li>
    <li>drawio-headless</li>
    <li>pandoc-drawio-filter</li>
    <li>pandoc-drawio-filter.dist</li>
    <li>python310Packages.mkdocs-drawio-exporter</li>
    <li>python310Packages.mkdocs-drawio-exporter.dist</li>
    <li>python311Packages.mkdocs-drawio-exporter</li>
    <li>python311Packages.mkdocs-drawio-exporter.dist</li>
  </ul>
</details>

File: pkgs/applications/graphics/drawio/default.nix
Comment (alyssais) [reply pkgs/applications/graphics/drawio/default.nix:25]: I find it a lot less readable, but I'm happy to defer to you as the original package maintainer.

File: pkgs/applications/graphics/drawio/default.nix
Comment (alyssais) [reply pkgs/applications/graphics/drawio/default.nix:1]: Could it be a mesa version mismatch between your system and the Nixpkgs you built the application from?  I see messages like that sometimes in that scenario.

**Comment by alyssais** [comment]
Not sure why the Darwin build is failing.  I've just marked it broken for now, since it didn't work before anyway.

File: pkgs/applications/graphics/drawio/default.nix
Comment (DarkOnion0) [reply pkgs/applications/graphics/drawio/default.nix:1]: Indeed, that's the case, I think that since the app is running well it's not a real issue.

**Review by DarkOnion0 — APPROVED** [review]

**Pull request merged**
